client/app/views/popover_screens/guests.coffee

```diff
@@ -158,31 +183,79 @@ module.exports = class GuestPopoverScreen extends EventPopoverScreenView
     # Handle guest addition. `userInfo` is passed when called by the typeahead.
     onNewGuest: (userInfo = null) ->
 
+        # Autocomplete was used.
         if userInfo? and typeof(userInfo) is "string"
-            [email, contactID] = userInfo.split(';')
+            [channel, contactID] = userInfo.split(';')
+        # Field was entered manually.
         else
-            email = @$('input[name="guest-name"]').val()
+            channel   = @$('input[name="guest-name"]').val()
             contactID = null
 
-        # An empty value should not be submitted.
-        email = email.trim()
-        if email.length > 0
+        # Determine if guest's "channel" of communication is the url of his cozy
+        # or his mail address.
+        if (channel.indexOf "@") < 0
+            cozy  = channel
+        else
+            email = channel
+            # An empty value should not be submitted.
+            email = email.trim()
+
+        if email?.length > 0 or cozy?.length > 0
             guests = @formModel.get('attendees') or []
-            if not _.findWhere(guests, email: email)
+
+            # Look for a duplicate:
+            # * another guest with the same email;
+            if email?
+                guestBisEmail = _.findWhere(guests, email: email)
+            # * another guest with the same cozy;
+            if cozy?
+                guestBisCozy  = _.findWhere(guests, cozy: cozy)
+
```

Comment:
I don't have an opinion on that either, I do not see much difference between the two options. Maybe one scenario is better for the user? @kelukelu ?